payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,116 @@
+package com.stripe.android.payments.core.authentication
+
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.Logger
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentBrowserAuthStarter
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.injection.AuthenticationModule
+import com.stripe.android.payments.core.injection.DaggerAuthenticationComponent
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarter
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Default registry to provide look ups for [IntentAuthenticator].
+ * Should be only accessed through [DefaultIntentAuthenticatorRegistry.getInstance].
+ */
+internal class DefaultIntentAuthenticatorRegistry @Inject internal constructor() :
+    IntentAuthenticatorRegistry {
+
+    // default authenticators always packaged with the SDK
+    @Inject
+    lateinit var webIntentAuthenticator: WebIntentAuthenticator
```

Comment:
Thanks for the heads up, I'm trying to reduce the scope of the PR by only moving code around without adding new features:)  so thought it would be easier to just directly declare the 3 deps inside the registry

Specifically, I left a TODO in [IntentAuthenticatorRegistry](https://github.com/stripe/stripe-android/pull/3774/files#diff-e80708393ba3326b4bf454b5048219b469f93ccd1a00299dadbc09ace79478be) to add the registration API(with dagger multibinding or reflection) in a follow up PR, where we might extract `Stripe3DS2Authenticator` into its own module and use the register API to register that. 

Added another TODO item to register the singleton instance when registration API is ready